Ska/Sun/sun.py

```diff
@@ -2,14 +2,29 @@
 """
 Utility for calculating sun position and pitch angle.
 """
+from pathlib import Path
+import numpy as np
+from math import cos, sin, acos, atan2, asin, pi, radians, degrees
+from astropy.table import Table
 
 from Quaternion import Quat
 from chandra_aca.transform import radec_to_eci
 from Chandra.Time import DateTime
-from math import cos, sin, acos, atan2, asin, pi, radians, degrees
-import numpy as np
 import Ska.quatutil
 
+ROLL_TABLE = Table.read(str(Path(__file__).parent / 'data' / 'pitch_rolldev.csv'),
```

Comment:
Do this as a `ska_helpers.util.LazyVal` instance so the table is read only on demand. 

I'm being more conscious of avoiding side-effects of importing modules, and `LazyVal` is an easy way to do that. It does have a syntactic disadvantage of having to write `ROLL_TABLE.val` later, but in this context that is not a problem.

Reply:
The `str` around `Path` is not needed.

Reply:
Regarding the "str" around "Path".  What is the nicest way to make this work with format='ascii.basic'?  ```     20     """
     21     file = Path(__file__).parent / 'data' / 'pitch_rolldev.csv'
---> 22     ROLL_TABLE = Table.read(file,
     23                             format='ascii.basic',
     24                             guess=False, delimiter=',')

/proj/sot/ska3/flight/lib/python3.8/site-packages/astropy/table/connect.py in __call__(self, *args, **kwargs)
     59         descriptions = kwargs.pop('descriptions', None)
     60 
---> 61         out = registry.read(cls, *args, **kwargs)
     62 
     63         # For some readers (e.g., ascii.ecsv), the returned `out` class is not

/proj/sot/ska3/flight/lib/python3.8/site-packages/astropy/io/registry.py in read(cls, format, cache, *args, **kwargs)
    518 
    519         reader = get_reader(format, cls)
--> 520         data = reader(*args, **kwargs)
    521 
    522         if not isinstance(data, cls):

/proj/sot/ska3/flight/lib/python3.8/site-packages/astropy/io/ascii/connect.py in io_read(format, filename, **kwargs)
     16         format = re.sub(r'^ascii\.', '', format)
     17         kwargs['format'] = format
---> 18     return read(filename, **kwargs)
     19 
     20 

/proj/sot/ska3/flight/lib/python3.8/site-packages/astropy/io/ascii/ui.py in read(table, guess, **kwargs)
    355             fast_reader_rdr = get_reader(**fast_kwargs)
    356             try:
--> 357                 dat = fast_reader_rdr.read(table)
    358                 _read_trace.append({'kwargs': copy.deepcopy(fast_kwargs),
    359                                     'Reader': fast_reader_rdr.__class__,

/proj/sot/ska3/flight/lib/python3.8/site-packages/astropy/io/ascii/fastbasic.py in read(self, table)
    107         self.kwargs['fast_reader'] = fast_reader
    108 
--> 109         self.engine = cparser.CParser(table, self.strip_whitespace_lines,
    110                                       self.strip_whitespace_fields,
    111                                       delimiter=self.delimiter,

astropy/io/ascii/cparser.pyx in astropy.io.ascii.cparser.CParser.__cinit__()

astropy/io/ascii/cparser.pyx in astropy.io.ascii.cparser.CParser.setup_tokenizer()

TypeError: Input "table" must be a file-like object, a string (filename or data), or an iterable```

Reply:
Argh, sorry about the `str`. I thought we had full support for `Path` everywhere, but apparently not. So just put the `str()` back. If you are feeling energetic you can open a bug report on astropy, this is definitely a bug.